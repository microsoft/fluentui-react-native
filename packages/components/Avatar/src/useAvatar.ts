import { ImageProps, ImageSourcePropType } from 'react-native';
import { AvatarProps, AvatarInfo, AvatarState, AvatarColors, AvatarSize } from './Avatar.types';
import { PresenceBadgeProps } from '@fluentui-react-native/badge';
import { titles } from './titles';
import { getHashCodeWeb } from './getHashCode';
import { createIconProps } from '@fluentui-react-native/interactive-hooks';
/**
 * Re-usable hook for FURN Avatar.
 * This hook configures Avatar props and state for FURN Avatar.
 *
 * @param props user props sent to FURN Avatar
 * @returns configured props and state for FURN Avatar
 */
export const useAvatar = (props: AvatarProps): AvatarInfo => {
  const {
    avatarColor,
    active,
    accessibilityLabel,
    accessibilityRole,
    activeAppearance,
    badge,
    idForColor,
    image,
    imageUrl,
    initials,
    name,
    shape,
    transparentRing,
    icon,
    size,
    ...rest
  } = props;

  const showRing = active === 'active' && activeAppearance === 'ring';
  const showBadge = (!active || active === 'unset') && !!badge && !!badge.status;
  const accessibilityText = `${name || ''}${showBadge ? `, ${badge.status}` : ''}`;

  const badgeProps: PresenceBadgeProps = {
    size: 'small',
    ...badge,
  };

  const state: AvatarState = {
    showRing,
    transparentRing: !!transparentRing,
    showBadge,
  };

  let imageProps: ImageProps = {
    accessibilityLabel,
    source: imageUrl ? ({ uri: imageUrl } as ImageSourcePropType) : undefined,
  };

  if (image && image.source) {
    imageProps = {
      ...image,
      accessibilityLabel: image.accessibilityLabel ?? accessibilityLabel,
    };
  }

  const _initials = initials || getInitials(name);
  const avatarColorsIdx = getHashCodeWeb(idForColor ?? name ?? '') % AvatarColors.length;
  const _avatarColor = avatarColor === 'colorful' ? AvatarColors[avatarColorsIdx] : avatarColor;

  let iconProps = createIconProps(icon);
  const isFontIcon = !!(iconProps && iconProps.fontSource);
  if (isFontIcon) {
    const { fontSource, ...restIconProps } = iconProps;

    iconProps = {
      fontSource: {
        fontSize: fontSource.fontSize ?? getFontIconSize(size),
        ...iconProps.fontSource,
      },
      ...restIconProps,
    };
  }

  return {
    props: {
      accessible: true,
      accessibilityLabel: accessibilityLabel || accessibilityText,
      accessibilityRole: accessibilityRole ?? 'image',
      active,
      activeAppearance,
      avatarColor: _avatarColor,
      badge: badgeProps,
      badgeStatus: badge?.status,
      initials: _initials,
      icon: iconProps,
      image: imageProps,
      outOfOffice: badge?.outOfOffice,
      shape: shape ?? 'circular',
      size,
      ...rest,
    },
    state: {
      ...state,
    },
  };
};

/**
 * A function which takes a name and returns initials
 *
 * @param name
 * @returns initials. First letter of the word if only one word was provided.
 * First letters of the first and last words if more words were provided.
 * Words in braces, titles, special characters, parantheses and dashes should be ignored.
 */
export const getInitials = (name: string): string => {
  if (!name) {
    return '';
  }
  const ARABIC_ASIAN_REGEXP = new RegExp(
    '[\u0600-\u06FF\u0750-\u077F\u08A0-\u08FF\u1100-\u11FF\u3130-\u318F\uA960-\uA97F\uAC00-\uD7AF\uD7B0-\uD7FF\u3040-\u309F\u30A0-\u30FF\u3400-\u4DBF\u4E00-\u9FFF\uF900-\uFAFF\uD840-\uD869]',
    'g',
  );
  const words = removeRedundantCharacters(name);
  const wordsLength = words.length;
  const lastWordIdx = wordsLength - 1;
  const firstLetter = words[0].charAt(0).toUpperCase();
  const lastLetter = wordsLength > 1 ? words[lastWordIdx].charAt(0).toUpperCase() : '';
  const initials = `${firstLetter}${ARABIC_ASIAN_REGEXP.test(name) ? '' : lastLetter}`;
  return initials;
};

/**
 * A function which takes a name and returns array of valid words
 *
 * @param name
 * @returns array of words without phone numbers and special characters.
 */
export const removeRedundantCharacters = (name: string): string[] => {
  if (!name) {
    return [];
  }

  const MAX_NAME_LENGTH = 100;
  const _name = name.length > MAX_NAME_LENGTH ? name.trim().substring(0, MAX_NAME_LENGTH) : name;

  const WORDS_IN_BRACES_REGEXP = new RegExp('[(\\[\\{][^\\)\\]\\}]*[\\)\\]\\}]', 'g');
  const PHONE_NUMBER_REGEXP = new RegExp('(\\+|(\\d|\\s))', 'g');
  const SPECIAL_CHARACTERS_REGEXP = new RegExp('[!"#\'$%&*+,-./:;>=<?@^_`|~¡¢£¤¥¦§¨©ª«¬®ˉ°±²¼½¾¿×÷]', 'g');
  const words = _name
    .replace(WORDS_IN_BRACES_REGEXP, '')
    .replace(PHONE_NUMBER_REGEXP, ' ')
    .replace(SPECIAL_CHARACTERS_REGEXP, '')
    .trim()
    .split(' ');
  return removeTitlesFromName(words);
};

/**
 * A function which verifies whether words array contains title
 *
 * @param words array of words in name
 * @returns array of words without titles
 */
export const removeTitlesFromName = (words: string[]): string[] => {
  return words.filter((word) => !titles.has(word));
};

function getFontIconSize(size: AvatarSize) {
  if (size >= 25 && size <= 47) {
    return 20;
  } else if (size === 48) {
    return 24;
  } else if (size === 56) {
    return 28;
  } else if (size >= 57 && size <= 72) {
    return 32;
  } else if (size >= 73) {
    return 48;
  }
  return 16;
}
