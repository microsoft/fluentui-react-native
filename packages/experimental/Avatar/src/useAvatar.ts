import { AvatarProps, AvatarInfo, AvatarState } from './Avatar.types';
import { getPresenceIconSource } from './Avatar.helpers';

/**
 * Re-usable hook for FURN Avatar.
 * This hook configures Avatar props and state for FURN Avatar.
 *
 * @param props user props sent to FURN Avatar
 * @returns configured props and state for FURN Avatar
 */
export const useAvatar = (props: AvatarProps): AvatarInfo => {
  const { imageUrl, imageDescription, initials, presence, isOutOfOffice, ring, ...rest } = props;

  const personaPhotoSource =
    imageUrl === undefined
      ? undefined
      : {
          uri: imageUrl,
        };

  const iconSource = presence === undefined ? undefined : getPresenceIconSource(presence, isOutOfOffice || false);
  const showRing = !!ring;
  const transparentRing = !!ring?.transparent;

  const state: AvatarState = {
    iconSource,
    personaPhotoSource,
    showRing,
    transparentRing,
  };

  return {
    props: {
      children: initials,
      accessibilityLabel: imageDescription,
      ...rest,
    },
    state: {
      ...state,
    },
  };
};
