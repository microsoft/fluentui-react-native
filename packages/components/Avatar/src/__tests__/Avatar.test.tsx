import * as React from 'react';

import * as renderer from 'react-test-renderer';

import { Avatar } from '..';
import { getInitials, resolveColorfulToSpecificColor } from '../useAvatar';

const emptyData = [
  [undefined, ''],
  [null, ''],
  ['', ''],
];

const testData = [
  ['Marie', 'M'],
  ['Aadi KaPoor', 'AK'],
  ['Marie Beaudouin', 'MB'],
  ['Aadi Meni KaPoor', 'AK'],
  ['Will Little (Teams)', 'WL'],
  ['Will Little-Tanaka', 'WL'],
  ['-Aadi Kapoor', 'AK'],
  ['Richard 23 Feynman', 'RF'],
  ['* Richard *', 'R'],
  ['Richard Feynman [Teams],', 'RF'],
  ['Richard Feynman {Teams}', 'RF'],
  ['-Test   str  with *spaces', 'TS'],
  ['Second, First', 'SF'],
];

const namesWithTitles = [
  ['Mr Feynman', 'F'],
  ['Dr Richard Feynman', 'RF'],
  ['Aadi Meni Ka Jr', 'AK'],
  ['Consul General Richard Faynman', 'RF'],
  ['Major General Victor', 'V'],
];

const languageTestData = [
  ['خسرو رحیمی', 'خ'], // Arabic
  ['桂英', '桂'], // Chinese
  ['佳', '佳'],
  ['宋智洋', '宋'],
  ['강현', '강'], // Korean
  ['최종래', '최'],
  ['남궁 성종', '남'],
  ['松田', '松'], // Japanese
  ['海野', '海'],
  ['かり', 'か'],
  ['Илон Маск', 'ИМ'], // Cyrillic
  ['Írissa Þórðardóttir', 'ÍÞ'], // non-ASCII characters
  ['Øyvind Åsen', 'ØÅ'],
  ['ثابت بن قره', 'ث'], // Thābit ibn Qurra - result in PCX - ثابت
  ['שירה לוי', 'של'], // Hebrew - Shira Levi
];

const phoneNumbersAndDigits = [
  ['12345678', ''],
  ['+47 12 34 56 78 (X 5678)', ''],
  ['47 12 34', ''],
  ['47 12', ''],
  ['James Ext 2', 'JE'],
  ['1', ''],
  ['A2', 'A'],
];

const multiWordTitlesData = [
  ['Air Comm', ''],
  ['Brig & Mrs', ''],
  ['Brig Gen', ''],
  ['Commander & Mrs', ''],
  ['Consul General', ''],
  ['Dr & Mrs', ''],
  ['Lord & Lady', ''],
  ['Major General', ''],
  ['Mr & Dr', ''],
  ['Mr & Mrs', ''],
  ['Mr & Ms', ''],
  ['Prof & Dr', ''],
  ['Prof & Mrs', ''],
  ['Rev & Mrs', ''],
  ['Rev Canon', ''],
  ['Rev Dr', ''],
  ['Sir & Lady', ''],
  ['The Rt Hon Lord Victor', 'V'],
];

const edgeCaseData = [
  ['4lex 5loo', 'LL'],
  [' !@#$%^&*()=+ (Alpha) David   (The man) `~<>,./?[]{}|   Goff   (Gamma)    ', 'DG'],
  ['+1 (555) 123-4567 ext.4567', 'E'],
  ['1 Ext 2', 'E'],
  ['1x1', 'X'],
];

describe('getInitials method', () => {
  it.each(emptyData)("returns an empty string for '%s'", (text, expected) => {
    expect(getInitials(text)).toBe(expected);
  });
  it.each(testData)("render correct initials for input '%s'", (text, expected) => {
    expect(getInitials(text)).toBe(expected);
  });
  it.each(namesWithTitles)("render initials for names with titles: '%s'", (text, expected) => {
    expect(getInitials(text)).toBe(expected);
  });
  it.each(languageTestData)("render initials for different languages: '%s'", (text, expected) => {
    expect(getInitials(text)).toBe(expected);
  });
  it.each(phoneNumbersAndDigits)("returns an empty string for phone numbers: '%s'", (text, expected) => {
    expect(getInitials(text)).toBe(expected);
  });
  it.each(edgeCaseData)("render initials for edge cases: '%s'", (text, expected) => {
    expect(getInitials(text)).toBe(expected);
  });
  it.each(multiWordTitlesData)("render initials for multi-word titles: '%s'", (text, expected) => {
    expect(getInitials(text)).toBe(expected);
  });
});

describe('resolveColorfulToSpecificColor method', () => {
  it('Name = Satya Nadella', () => {
    expect(resolveColorfulToSpecificColor(undefined, 'Satya Nadella')).toBe('lilac');
  });
  it('Name = Bill Gates', () => {
    expect(resolveColorfulToSpecificColor(undefined, 'Bill Gates')).toBe('plum');
  });
  it('Undefined case', () => {
    expect(resolveColorfulToSpecificColor(undefined, undefined)).toBe('darkRed');
  });
  it('Id = a', () => {
    expect(resolveColorfulToSpecificColor('a', undefined)).toBe('brass');
  });
  it('Id = x', () => {
    expect(resolveColorfulToSpecificColor('x', undefined)).toBe('darkRed');
  });
});

describe('Avatar component tests', () => {
  it('Avatar default', () => {
    const tree = renderer.create(<Avatar />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Avatar circular', () => {
    const tree = renderer.create(<Avatar shape="circular" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Avatar square', () => {
    const tree = renderer.create(<Avatar shape="square" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Avatar badge', () => {
    const tree = renderer.create(<Avatar badge={{ status: 'available', outOfOffice: false }} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Avatar ring', () => {
    const tree = renderer.create(<Avatar active="active" activeAppearance="ring" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
