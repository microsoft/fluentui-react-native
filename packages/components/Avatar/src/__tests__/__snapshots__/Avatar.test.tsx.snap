// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component tests Avatar badge 1`] = `
<View
  accessibilityLabel=", available"
  accessibilityRole="image"
  accessible={true}
  badgeStatus="available"
  icon={null}
  image={
    {
      "accessibilityLabel": "",
      "source": undefined,
    }
  }
  initials=""
  outOfOffice={false}
  shape="circular"
  style={
    {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "aspectRatio": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "minHeight": 24,
      "minWidth": 24,
      "opacity": 1,
    }
  }
>
  <View
    accessible={false}
    style={
      {
        "alignItems": "center",
        "aspectRatio": 1,
        "backgroundColor": "#e6e6e6",
        "borderColor": "#d1d1d1",
        "borderRadius": 9999,
        "borderWidth": 0,
        "justifyContent": "center",
        "minHeight": 24,
        "minWidth": 24,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="16"
      bbWidth="16"
      color="#616161"
      focusable={false}
      height={16}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 16,
            "width": 16,
          },
        ]
      }
      tintColor="#616161"
      vbHeight={20}
      vbWidth={20}
      width={16}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGPath
          d="M10 2C7.79086 2 6 3.79086 6 6C6 8.20914 7.79086 10 10 10C12.2091 10 14 8.20914 14 6C14 3.79086 12.2091 2 10 2ZM7 6C7 4.34315 8.34315 3 10 3C11.6569 3 13 4.34315 13 6C13 7.65685 11.6569 9 10 9C8.34315 9 7 7.65685 7 6ZM5.00873 11C3.90315 11 3 11.8869 3 13C3 14.6912 3.83281 15.9663 5.13499 16.7966C6.41697 17.614 8.14526 18 10 18C11.8547 18 13.583 17.614 14.865 16.7966C16.1672 15.9663 17 14.6912 17 13C17 11.8956 16.1045 11 15 11L5.00873 11ZM4 13C4 12.4467 4.44786 12 5.00873 12L15 12C15.5522 12 16 12.4478 16 13C16 14.3088 15.3777 15.2837 14.3274 15.9534C13.2568 16.636 11.7351 17 10 17C8.26489 17 6.74318 16.636 5.67262 15.9534C4.62226 15.2837 4 14.3088 4 13Z"
          fill={
            {
              "type": 2,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    accessible={true}
    iconPosition="before"
    position="absolute"
    style={
      {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "aspectRatio": 1,
        "backgroundColor": "#ffffff",
        "borderColor": "#ffffff",
        "borderRadius": 9999,
        "borderWidth": 1,
        "bottom": -1,
        "flexDirection": "row",
        "height": 8,
        "justifyContent": "center",
        "paddingHorizontal": 0,
        "position": "absolute",
        "right": -1,
        "width": 8,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="6"
      bbWidth="6"
      fill="none"
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "aspectRatio": 1,
            "color": "#13a10e",
            "height": 6,
            "width": 6,
          },
          {
            "flex": 0,
            "height": 6,
            "width": 6,
          },
        ]
      }
      tintColor="#13a10e"
      vbHeight={16}
      vbWidth={16}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M8 16A8 8 0 108 0a8 8 0 000 16zm3.7-9.3l-4 4a1 1 0 01-1.41 0l-2-2a1 1 0 111.42-1.4L7 8.58l3.3-3.3a1 1 0 011.4 1.42z"
          fill={
            {
              "type": 2,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`Avatar component tests Avatar circular 1`] = `
<View
  accessibilityLabel=""
  accessibilityRole="image"
  accessible={true}
  icon={null}
  image={
    {
      "accessibilityLabel": "",
      "source": undefined,
    }
  }
  initials=""
  shape="circular"
  style={
    {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "aspectRatio": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "minHeight": 24,
      "minWidth": 24,
      "opacity": 1,
    }
  }
>
  <View
    accessible={false}
    style={
      {
        "alignItems": "center",
        "aspectRatio": 1,
        "backgroundColor": "#e6e6e6",
        "borderColor": "#d1d1d1",
        "borderRadius": 9999,
        "borderWidth": 0,
        "justifyContent": "center",
        "minHeight": 24,
        "minWidth": 24,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="16"
      bbWidth="16"
      color="#616161"
      focusable={false}
      height={16}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 16,
            "width": 16,
          },
        ]
      }
      tintColor="#616161"
      vbHeight={20}
      vbWidth={20}
      width={16}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGPath
          d="M10 2C7.79086 2 6 3.79086 6 6C6 8.20914 7.79086 10 10 10C12.2091 10 14 8.20914 14 6C14 3.79086 12.2091 2 10 2ZM7 6C7 4.34315 8.34315 3 10 3C11.6569 3 13 4.34315 13 6C13 7.65685 11.6569 9 10 9C8.34315 9 7 7.65685 7 6ZM5.00873 11C3.90315 11 3 11.8869 3 13C3 14.6912 3.83281 15.9663 5.13499 16.7966C6.41697 17.614 8.14526 18 10 18C11.8547 18 13.583 17.614 14.865 16.7966C16.1672 15.9663 17 14.6912 17 13C17 11.8956 16.1045 11 15 11L5.00873 11ZM4 13C4 12.4467 4.44786 12 5.00873 12L15 12C15.5522 12 16 12.4478 16 13C16 14.3088 15.3777 15.2837 14.3274 15.9534C13.2568 16.636 11.7351 17 10 17C8.26489 17 6.74318 16.636 5.67262 15.9534C4.62226 15.2837 4 14.3088 4 13Z"
          fill={
            {
              "type": 2,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`Avatar component tests Avatar default 1`] = `
<View
  accessibilityLabel=""
  accessibilityRole="image"
  accessible={true}
  icon={null}
  image={
    {
      "accessibilityLabel": "",
      "source": undefined,
    }
  }
  initials=""
  shape="circular"
  style={
    {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "aspectRatio": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "minHeight": 24,
      "minWidth": 24,
      "opacity": 1,
    }
  }
>
  <View
    accessible={false}
    style={
      {
        "alignItems": "center",
        "aspectRatio": 1,
        "backgroundColor": "#e6e6e6",
        "borderColor": "#d1d1d1",
        "borderRadius": 9999,
        "borderWidth": 0,
        "justifyContent": "center",
        "minHeight": 24,
        "minWidth": 24,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="16"
      bbWidth="16"
      color="#616161"
      focusable={false}
      height={16}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 16,
            "width": 16,
          },
        ]
      }
      tintColor="#616161"
      vbHeight={20}
      vbWidth={20}
      width={16}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGPath
          d="M10 2C7.79086 2 6 3.79086 6 6C6 8.20914 7.79086 10 10 10C12.2091 10 14 8.20914 14 6C14 3.79086 12.2091 2 10 2ZM7 6C7 4.34315 8.34315 3 10 3C11.6569 3 13 4.34315 13 6C13 7.65685 11.6569 9 10 9C8.34315 9 7 7.65685 7 6ZM5.00873 11C3.90315 11 3 11.8869 3 13C3 14.6912 3.83281 15.9663 5.13499 16.7966C6.41697 17.614 8.14526 18 10 18C11.8547 18 13.583 17.614 14.865 16.7966C16.1672 15.9663 17 14.6912 17 13C17 11.8956 16.1045 11 15 11L5.00873 11ZM4 13C4 12.4467 4.44786 12 5.00873 12L15 12C15.5522 12 16 12.4478 16 13C16 14.3088 15.3777 15.2837 14.3274 15.9534C13.2568 16.636 11.7351 17 10 17C8.26489 17 6.74318 16.636 5.67262 15.9534C4.62226 15.2837 4 14.3088 4 13Z"
          fill={
            {
              "type": 2,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`Avatar component tests Avatar ring 1`] = `
<View
  accessibilityLabel=""
  accessibilityRole="image"
  accessible={true}
  active="active"
  activeAppearance="ring"
  icon={null}
  image={
    {
      "accessibilityLabel": "",
      "source": undefined,
    }
  }
  initials=""
  shape="circular"
  style={
    {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "aspectRatio": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "minHeight": 32,
      "minWidth": 32,
      "opacity": 1,
    }
  }
>
  <View
    style={
      {
        "aspectRatio": 1,
        "backgroundColor": "transparent",
        "borderColor": "#00000000",
        "borderRadius": 9999,
        "borderStyle": "solid",
        "borderWidth": 2,
        "marginLeft": 2,
        "marginTop": 2,
        "minHeight": 32,
        "minWidth": 32,
      }
    }
  >
    <View
      accessible={false}
      style={
        {
          "alignItems": "center",
          "aspectRatio": 1,
          "backgroundColor": "#e6e6e6",
          "borderColor": "#d1d1d1",
          "borderRadius": 9999,
          "borderWidth": 0,
          "justifyContent": "center",
          "marginLeft": 2,
          "marginTop": 2,
          "minHeight": 24,
          "minWidth": 24,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="16"
        bbWidth="16"
        color="#616161"
        focusable={false}
        height={16}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 16,
              "width": 16,
            },
          ]
        }
        tintColor="#616161"
        vbHeight={20}
        vbWidth={20}
        width={16}
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
        >
          <RNSVGPath
            d="M10 2C7.79086 2 6 3.79086 6 6C6 8.20914 7.79086 10 10 10C12.2091 10 14 8.20914 14 6C14 3.79086 12.2091 2 10 2ZM7 6C7 4.34315 8.34315 3 10 3C11.6569 3 13 4.34315 13 6C13 7.65685 11.6569 9 10 9C8.34315 9 7 7.65685 7 6ZM5.00873 11C3.90315 11 3 11.8869 3 13C3 14.6912 3.83281 15.9663 5.13499 16.7966C6.41697 17.614 8.14526 18 10 18C11.8547 18 13.583 17.614 14.865 16.7966C16.1672 15.9663 17 14.6912 17 13C17 11.8956 16.1045 11 15 11L5.00873 11ZM4 13C4 12.4467 4.44786 12 5.00873 12L15 12C15.5522 12 16 12.4478 16 13C16 14.3088 15.3777 15.2837 14.3274 15.9534C13.2568 16.636 11.7351 17 10 17C8.26489 17 6.74318 16.636 5.67262 15.9534C4.62226 15.2837 4 14.3088 4 13Z"
            fill={
              {
                "type": 2,
              }
            }
            propList={
              [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</View>
`;

exports[`Avatar component tests Avatar square 1`] = `
<View
  accessibilityLabel=""
  accessibilityRole="image"
  accessible={true}
  icon={null}
  image={
    {
      "accessibilityLabel": "",
      "source": undefined,
    }
  }
  initials=""
  shape="square"
  style={
    {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "aspectRatio": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "minHeight": 24,
      "minWidth": 24,
      "opacity": 1,
    }
  }
>
  <View
    accessible={false}
    style={
      {
        "alignItems": "center",
        "aspectRatio": 1,
        "backgroundColor": "#e6e6e6",
        "borderColor": "#d1d1d1",
        "borderRadius": 4,
        "borderWidth": 0,
        "justifyContent": "center",
        "minHeight": 24,
        "minWidth": 24,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="16"
      bbWidth="16"
      color="#616161"
      focusable={false}
      height={16}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 16,
            "width": 16,
          },
        ]
      }
      tintColor="#616161"
      vbHeight={20}
      vbWidth={20}
      width={16}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGPath
          d="M10 2C7.79086 2 6 3.79086 6 6C6 8.20914 7.79086 10 10 10C12.2091 10 14 8.20914 14 6C14 3.79086 12.2091 2 10 2ZM7 6C7 4.34315 8.34315 3 10 3C11.6569 3 13 4.34315 13 6C13 7.65685 11.6569 9 10 9C8.34315 9 7 7.65685 7 6ZM5.00873 11C3.90315 11 3 11.8869 3 13C3 14.6912 3.83281 15.9663 5.13499 16.7966C6.41697 17.614 8.14526 18 10 18C11.8547 18 13.583 17.614 14.865 16.7966C16.1672 15.9663 17 14.6912 17 13C17 11.8956 16.1045 11 15 11L5.00873 11ZM4 13C4 12.4467 4.44786 12 5.00873 12L15 12C15.5522 12 16 12.4478 16 13C16 14.3088 15.3777 15.2837 14.3274 15.9534C13.2568 16.636 11.7351 17 10 17C8.26489 17 6.74318 16.636 5.67262 15.9534C4.62226 15.2837 4 14.3088 4 13Z"
          fill={
            {
              "type": 2,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
